core/Generics.carp

```diff
@@ -0,0 +1,106 @@
+;; The following generic functions make use of the interfaces
+
+(defmodule Generics
+  (defn one [] (inc (zero)))
+  (defn minus-one [] (dec (zero)))
+  (defn two [] (inc (one)))
+  (defn three [] (inc (two)))
+  (defn minus-two [] (* (two) (minus-one)))
+  (defn four [] (+ (two) (two)))
+  (defn five [] (+ (four) (one)))
+  (defn ten [] (+ (five) (five)))
+  (defn hundred [] (* (ten) (ten)))
+  (defn thousand [] (* (hundred) (ten)))
+  (defn hundred-thousand [] (* (hundred) (thousand)))
+  (defn million [] (* (thousand) (thousand)))
+  (defn minus-four [] (* (four) (minus-one)))
+  (defn twice [x] (* (two) x))
+  (defn halved [x] (/ x (two)))
+  (defn squared [x] (* x x))
+  (defn inverse [x] (/ (one) x))
+  (defn half [] (halved (one)))
+  (defn half-pi [] (halved pi))
+  (defn tau [] (twice pi))
+  (defn ninety [] (* (two) (* (* (three) (three)) (five))))
+  (defn one-hundred-eighty [] (* (two) (ninety)))
+  (defn small [] (inverse (hundred-thousand)))
+
+  (doc approx-margin "checks whether `x` and `y` are approximately equal within a `margin`.")
+  (defn approx-margin [x y margin]
+    (if (> x y)
+      (< (- x y) margin)
+      (< (- y x) margin)))
+
+  (doc approx-margin "checks whether `x` and `y` are approximately equal with a margin.
+
+The margin of error is 0.00001.")
+  (defn approx [a b]
+    (approx-margin a b (small)))
+
+)
+
+
+;; These are defined in the global scope.
+
+(use Generics)
```

Comment:
Since you're using the `use` command in the global scope, the Generics module will be included from here on, carrying into the user's session.
The best way to solve this is probably to use the `with` command instead, like this:
```
(with Generics
  <forms...>)
```
This will avoid the problem!

Reply:
Also, this feature might be completely undocumented... yikes :/

Reply:
> Also, this feature might be completely undocumented... yikes :/

I think it is! I saw it when digging through the source yesterday and couldn't recall any references to it in docs. Seems super useful!